src/jit/morph.cpp

```diff
@@ -4103,11 +4103,22 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #ifndef LEGACY_BACKEND
         if (argx->gtOper == GT_MKREFANY)
         {
-            NYI_X86("MKREFANY");
-
             // 'Lower' the MKREFANY tree and insert it.
             noway_assert(!reMorphing);
 
+#ifdef _TARGET_X86_
+
+            // Build the mkrefany as a GT_FIELD_LIST
```

Comment:
It's not the right approach for x64/Windows because it has to be passed by ref, but for other targets I think we should switch to this. However, I didn't want to perturb other targets with this.